browser/farbling/brave_dark_mode_fingerprint_protection_browsertest.cc

```diff
@@ -92,6 +92,36 @@ class BraveDarkModeFingerprintProtectionTest : public InProcessBrowserTest {
         content_settings(), ControlType::DEFAULT, top_level_page_url_);
   }
 
+  void SetDarkMode(bool dark_mode) {
+    test_theme_.SetDarkMode(dark_mode);
+    browser()
+        ->tab_strip_model()
+        ->GetActiveWebContents()
+        ->OnWebPreferencesChanged();
+  }
+
+  bool IsReportingDarkMode() {
+    bool light_mode_result = false;
+    EXPECT_TRUE(content::ExecuteScriptAndExtractBool(
+        contents(), base::StringPrintf(kMatchDarkModeFormatString, "light"),
```

Comment:
`content::EvalJs` and `content::JsReplace` can be used to simplify this.

Reply:
So, this one is unfortunately is not a good candidate for `EvalJs` from what I can see due to 2 things:
1. `JsReplace` escapes the args with `"`, so the media query turns into `matchMedia('(prefers-color-scheme: "light")').matches`, which doesn't match.
2. One of the tests uses `brave://settings` page, which doesn't work with `EvalJs` due to CSP. There was some discussion on fixing it https://groups.google.com/a/chromium.org/g/content-owners/c/52CC64X4CIw but there is https://bugs.chromium.org/p/chromium/issues/detail?id=1259739 that seems to indicate that it hasn't been...

So, I ended up not changing this, but did add an additional check (in aa036753cc79ed84c9520582889147576952e5e8) that was missing.